transaction/core/src/validation/validate.rs

```diff
@@ -302,16 +302,16 @@ fn validate_membership_proofs(
                 return Err(TransactionValidationError::InvalidLedgerContext);
             }
             Ok(derived_proof) => {
-                match derived_proof.elements.last() {
-                    None => {
+                match crate::membership_proofs::compute_implied_merkle_root(&derived_proof) {
```

Comment:
this had to change because the previous code assumes that `elements.last()` is the merkle root value given by the proof, but after this revision, the last element is actually the last thing we fold in, we have to call `compute_implied_merkle_root` to get the merkle root element